src/sentry/monitors/endpoints/organization_monitor_index_stats.py

```diff
@@ -0,0 +1,128 @@
+from __future__ import annotations
+
+from collections import OrderedDict, defaultdict
+from datetime import datetime, timedelta
+from typing import List, MutableMapping
+
+from django.db.models import Count, DateTimeField, Func
+from django.db.models.functions import Extract
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.base import StatsMixin, region_silo_endpoint
+from sentry.api.bases.organization import OrganizationEndpoint
+from sentry.api.helpers.environments import get_environments
+from sentry.monitors.models import CheckInStatus, MonitorCheckIn
+from sentry.utils.dates import to_timestamp
+
+
+def normalize_to_epoch(timestamp: datetime, seconds: int):
+    """
+    Given a ``timestamp`` (datetime object) normalize to an epoch timestamp.
+
+    i.e. if the rollup is minutes, the resulting timestamp would have
+    the seconds and microseconds rounded down.
+    """
+    epoch = int(to_timestamp(timestamp))
+    return epoch - (epoch % seconds)
+
+
+@region_silo_endpoint
+class OrganizationMonitorIndexStatsEndpoint(OrganizationEndpoint, StatsMixin):
+    # TODO(epurkhiser): probably convert to snuba
+    def get(self, request: Request, organization) -> Response:
+        args = self._parse_args(request)
+
+        start = normalize_to_epoch(args["start"], args["rollup"])
+        end = normalize_to_epoch(args["end"], args["rollup"])
+
+        monitor_ids_str: List[str] = request.GET.getlist("monitor")
```

Comment:
should we be using slug here, im not sure if frontend has access to the actual id

Reply:
You're right we don't return that. I'll update this to use slug.

Also this was a unchecked access vulnerability so 👍